### PR TITLE
Rework config file to make it more useful

### PR DESCRIPTION
**Pull request by Eragos**
* Remove config.js and add config.js.dist instead
* Split config file into categories
* Add hard-codet settings to config file #3 
* Add config.js, package-lock.json to .gitignore
* Remove package-lock.json from repository

Todo:
Maybe add envoirement variables to the new config options

### CODE REVIEW

File: config.js.dist [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module.exports = {
+  /**
+   * Discord Bot settings. These are mandatory and will affect what the bot
+   *   posts about, and where it posts to.
+   */
+  config: {
```
Comment (ryan-rowland) [inline config.js.dist]: We don't need this nested extra layer, `config`. Can we move the comment above to the very top of the file, and remove `config`, moving all of the child properties up a level? Eg:
```
module.exports = {
  discord: { ... },
  guild: { ... },
  ...
```

File: config.js.dist [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    },
+    battle: {
+      // Min players to report as battle
+      min_players: 10,
+      // Min guild players to report as battle
+      min_relevant_players: 3
+    },
+    kill: {
+      // Min killfame to report kill
+      min_fame: 25000
```
Comment (ryan-rowland) [inline config.js.dist]: Can we change `min_players`, `min_relevant_players` and `min_fame` to snakeCase for consistency with the other settings?

**Comment by Eragos** [comment]
I just push the requested Changes to the PR.

**Comment by ryan-rowland** [comment]
LGTM. 👍 Thanks @Eragos !

**Pull request merged**